src/sentry/snuba/discover.py

```diff
@@ -1325,6 +1325,69 @@ def get_facets(
     return results
 
 
+def spans_histogram_query(
+    span,
+    user_query,
+    params,
+    num_buckets,
+    precision=0,
+    min_value=None,
+    max_value=None,
+    data_filter=None,
+    referrer=None,
+    group_by=None,
+    order_by=None,
+    limit_by=None,
+    extra_snql_condition=None,
+    normalize_results=True,
+):
+    multiplier = int(10 ** precision)
+    if max_value is not None:
+        max_value -= 0.1 / multiplier
+
+    # TODO add min max calculation after
+    # min_value, max_value = find_histogram_min_max(
+    #     fields, min_value, max_value, user_query, params, data_filter, use_snql
+    # )
+
+    key_column = None
+    field_names = []
+    histogram_rows = None
+
+    # TODO calculate histogram_params
+    # histogram_params = find_histogram_params(num_buckets, min_value, max_value, multiplier)
+    histogram_params = HistogramParams(num_buckets=100, bucket_size=1, start_offset=0, multiplier=1)
+    histogram_column = get_span_histogram_column(span, histogram_params)
+
+    builder = HistogramQueryBuilder(
+        num_buckets,
+        histogram_column,
+        histogram_rows,
+        histogram_params,
+        key_column,
+        field_names,
+        group_by,
+        # Arguments for QueryBuilder
+        Dataset.Discover,
+        params,
+        query=user_query,
+        # TODO figure out what should be passed to selected_columns
+        selected_columns=["spans.exclusive_time_32"],
```

Comment:
not sure what this is used for. could this be an arbitrary string since we're using this to calculate `group` 🤔

Reply:
It's for multi-histograms, it looks like perhaps it's required since `len` is called on it and is used to determine the limit for the query, but the value itself might not matter.

Reply:
This was used for multi histograms, and in those cases, we would pass the key columns (e.g. `measurements.key`). In this case, I'd leave this empty since the query will group by the columns you add here which is not what you want.

Reply:
ah I see 👍 updated passed in value